plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableDataStoreTests.php

```diff
@@ -3085,6 +3085,21 @@ function () use ( &$count ) {
 			}
 		);
 
+		/**
+		 * fix for previously flaky test:
+		 * less than a second of time passes while the following saves happen.
+		 */
+		$datetime = new DateTime( 'now', new DateTimeZone( 'UTC' ) );
+		$now      = $datetime->format( 'Y-m-d H:i:s' );
+		$this->reset_legacy_proxy_mocks();
+		$this->register_legacy_proxy_function_mocks(
+			array(
+				'current_time' => function( $type, $gmt ) {
+					return $now;
+				},
```

Comment:
This is incorrect: `$now` is interpreted as a variable defined inside the function and thus the value actually returned is undefined (probably `null`). It should have an `use` clause:

```php
'current_time' => function( $type, $gmt ) use ( $now ) {
	return $now;
},
```

Or alternatively, the new lambda syntax introduced in PHP 7.4 could be used which is more concise (and doesn't require `use`s):

```php
'current_time' => fn( $type, $gmt ) => $now
```

Reply:
Great catch, thank you @Konamiman! Looking into it further I've discovered that the `current_time` mock never gets called in the first place. I'll explore this some more.

Reply:
Addressed this. ✅